apps/storefront/src/pages/OrderDetail/index.test.tsx

```diff
@@ -1215,4 +1253,806 @@ describe('when a personal customer visits an order', () => {
+
+      const shoppingList = buildCustomerShoppingListNodeWith({
+        node: { id: '992', name: 'Foo Bar Shopping List' },
+      });
+
+      server.use(
+        graphql.query('CustomerShoppingLists', () =>
+          HttpResponse.json(
+            buildCustomerShoppingListResponseWith({
+              data: { customerShoppingLists: { totalCount: 1, edges: [shoppingList] } },
+            }),
+          ),
+        ),
+        graphql.query('SearchProducts', () => HttpResponse.json({ data: { productsSearch: [] } })),
+        graphql.mutation('AddItemsToCustomerShoppingList', ({ query }) =>
+          HttpResponse.json(addItemsToCustomerShoppingList(query)),
+        ),
+      );
+
+      when(addItemsToCustomerShoppingList)
+        .calledWith(
+          stringContainingAll(
+            'productId: 123,',
+            'variantId: 456,',
+            'quantity: 12,',
+            '{optionId: "attribute[22]", optionValue: "bar" },',
+            'shoppingListId: 992,',
+          ),
+        )
+        .thenReturn({});
+
+      server.use(
+        graphql.query('GetCustomerOrderStatuses', () =>
+          HttpResponse.json(buildCustomerOrderStatusesWith('WHATEVER_VALUES')),
+        ),
+        graphql.query('AddressConfig', () =>
+          HttpResponse.json(buildAddressConfigResponseWith('WHATEVER_VALUES')),
+        ),
+        graphql.query('GetCustomerOrder', () =>
+          HttpResponse.json(
+            buildCustomerOrderResponseWith({
+              data: { customerOrder: { products: [screamCanister, laughCanister] } },
+            }),
+          ),
+        ),
+        graphql.query('CustomerShoppingLists', () =>
+          HttpResponse.json(
+            buildCustomerShoppingListResponseWith({
+              data: { customerShoppingLists: { totalCount: 1, edges: [shoppingList] } },
+            }),
+          ),
+        ),
+        graphql.query('SearchProducts', () => HttpResponse.json({ data: { productsSearch: [] } })),
+        graphql.mutation('AddItemsToCustomerShoppingList', ({ query }) =>
+          HttpResponse.json(addItemsToCustomerShoppingList(query)),
+        ),
+      );
+
+      renderWithProviders(<OrderDetails />, {
+        preloadedState,
+        initialGlobalContext: { shoppingListEnabled: true },
+      });
+
+      await waitForElementToBeRemoved(() => screen.queryAllByRole('progressbar'));
+
+      await userEvent.click(screen.getByRole('button', { name: 'ADD TO SHOPPING LIST' }));
+
+      const dialog = await screen.findByRole('dialog', { name: 'Add to shopping list' });
+
+      expect(
+        within(dialog).getByText('Select products and quantity to add to shopping list'),
+      ).toBeInTheDocument();
+
+      const productGroup = within(dialog).getByRole('group', { name: 'Laugh Canister' });
+
+      await userEvent.click(within(productGroup).getByRole('checkbox'));
+      await userEvent.type(within(productGroup).getByRole('spinbutton'), '2');
```

Comment:
as above, the expected `12` is a little buried at line 1778, so it'd be useful to remind people how this `2` relates